src/OrleansAzureUtils/MultiClusterNetwork/GossipTableInstanceManager.cs

```diff
@@ -0,0 +1,327 @@
+
+            return queryResults.Select(tuple => tuple.Item1).ToList();
+        }
+
+        internal async Task<GossipTableEntry> ReadConfigurationEntryAsync()
+        {
+            var result = await storage.ReadSingleTableEntryAsync(this.GlobalServiceId, GossipTableEntry.CONFIGURATION_ROW).ConfigureAwait(false);
+            return result != null ? result.Item1 : null;
+        }
+
+        internal async Task<GossipTableEntry> ReadGatewayEntryAsync(GatewayEntry gateway)
+        {
+            var result = await storage.ReadSingleTableEntryAsync(this.GlobalServiceId, GossipTableEntry.ConstructRowKey(gateway.SiloAddress, gateway.ClusterId)).ConfigureAwait(false);
+            return result != null ? result.Item1 : null;
+        }
+
+        internal async Task<bool> TryCreateConfigurationEntryAsync(MultiClusterConfiguration configuration)
+        {
+            if (configuration == null) throw new ArgumentNullException("configuration");
+
+            var entry = new GossipTableEntry
+            {
+                PartitionKey = GlobalServiceId,
+                RowKey = GossipTableEntry.CONFIGURATION_ROW,
+                GossipTimestamp = configuration.AdminTimestamp,
+                Clusters = string.Join(GossipTableEntry.ClustersListSeparator, configuration.Clusters),
+                Comment = configuration.Comment ?? ""
+            };
+
+            return (await TryCreateTableEntryAsync(entry).ConfigureAwait(false) != null);
+        }
+
+        internal async Task<bool> TryUpdateConfigurationEntryAsync(MultiClusterConfiguration configuration, GossipTableEntry entry, string eTag)
+        {
+            if (configuration == null) throw new ArgumentNullException("configuration");
+
+            //Debug.Assert(entry.ETag == eTag);
+            //Debug.Assert(entry.PartitionKey == GlobalServiceId);
+            //Debug.Assert(entry.RowKey == GossipTableEntry.CONFIGURATION_ROW);
+
+            entry.GossipTimestamp = configuration.AdminTimestamp;
+            entry.Clusters = string.Join(GossipTableEntry.ClustersListSeparator, configuration.Clusters);
+            entry.Comment = configuration.Comment ?? "";
+
+            return (await TryUpdateTableEntryAsync(entry, eTag).ConfigureAwait(false) != null);
+        }
+
+        internal async Task<bool> TryCreateGatewayEntryAsync(GatewayEntry entry)
+        {
+            var row = new GossipTableEntry()
+            {
+                PartitionKey = GlobalServiceId,
+                RowKey = GossipTableEntry.ConstructRowKey(entry.SiloAddress, entry.ClusterId),
+                Status = entry.Status.ToString(),
+                GossipTimestamp = entry.HeartbeatTimestamp
+            };
+
+            return (await TryCreateTableEntryAsync(row).ConfigureAwait(false) != null);
+        }
+
+
+        internal async Task<bool> TryUpdateGatewayEntryAsync(GatewayEntry entry, GossipTableEntry row, string eTag)
+        {
+            //Debug.Assert(row.ETag == eTag);
```

Comment:
Ok. Though I think seeing the asserts may be helpful for understanding the preconditions of the method. I commented them out only because Orleans codebase does not have asserts.